proposals/public-suffix.md

```diff
@@ -0,0 +1,555 @@
+# Proposal: Public Suffix API
+
+**Summary**
+
+API to obtain the *registrable domain / eTLD+1 (effective Top Level Domain+1)*
+from a domain name.
+
+**Document Metadata**
+
+**Author:** [Francis McKenzie](https://github.com/mckenfra)
+
+**Sponsoring Browser:** Mozilla Firefox
+
+**Contributors:** N/A
+
+**Created:** 2024-08-19
+
+**Related Issues:** [#231](https://github.com/w3c/webextensions/issues/231)
+
+## Motivation
+
+### Objective
+
+This API enables developers to obtain the *registrable domain / eTLD+1*
+from a domain name. This functionality is already implemented for internal
+use by all the major browsers. Therefore the effect of this API is to expose
+existing built-in browser functionality to extensions developers.
+
+The primary objective of this API is to eliminate the possibility of inconsistencies
+between the host browser and hosted extensions when deriving
+*registrable domain / eTLD+1*s from domain names.
+
+Secondary objectives of this API are to:
+
+1. Improve extension developer experience by reducing complexity and maintenance overhead,
+since developers will no longer need to roll their own solutions for obtaining and parsing
+the [Public Suffix List (PSL)](https://publicsuffix.org/list/).
+2. Reduce extensions' resource usage (CPU, memory, disk space), since extensions
+will no longer be duplicating work already done by the host browser.
+
+### Features of the PSL
+
+In order to set out the use cases fully, it is first necessary to describe certain
+relevant features of the PSL.
+
+#### 1. ICANN vs Private
+
+The PSL it is divided into two sections: ICANN suffixes and Private (i.e. non-ICANN)
+suffixes.
+
+##### Examples
+
+| ICANN Suffixes                | Private Suffixes (Non-ICANN)  |
+|-------------------------------|------------------------------ |
+| com                           | blogspot.com                  |
+|                               | members.linode.com            |
+|                               | simplesite.com                |
+|                               |                               |
+| co.uk                         | blogspot.co.uk                |
+|                               | wellbeingzone.co.uk           |
+|                               | vm.bytemark.co.uk             |
+
+The suffixes in the PSL represent boundaries between registrar-type organizations
+and their clients. The "owner" of the suffix is one organization, and the "owner"
+of a domain label immediately preceeding the suffix is a different organization.
+
+In the case of suffixes in the ICANN section, the "owner" of a suffix is ICANN /
+an affiliated registrar. In the case of suffixes in the Private section, the
+"owner" of a suffix is a private organization offering a service for clients
+to take ownership of subdomains underneath its own ICANN-suffixed domain.
+
+#### 2. Multiple Suffixes per Domain
+
+If a domain name ends with a suffix listed in the Private section of the PSL,
```

Comment:
Maybe stress that in almost all cases you want to use the full PSL. If you're trying to attribute abuse at some point you want to go from "a.foo.com" and "b.foo.com" and "c.foo.com" are all distributing malware to "foo.com" is distributing malware. That's the only reason I can think of, though and maybe that should instead be a generic "attribute to all higher labels".